### PR TITLE
Use new RubyStringBuilder.str api to create inspected string for Exception

### PR DESCRIPTION
**Pull request by jlahtinen**
pull request fix for #5923

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyException.java [hunk 1]
```diff
@@ -302,10 +303,9 @@ public RubyString inspect(ThreadContext context) {
 
         if (exception.isEmpty()) return context.runtime.newString(rubyClass);
 
-        return RubyString.newString(context.runtime,
-                new StringBuilder(2 + rubyClass.length() + 2 + exception.size() + 1).
-                    append("#<").append(rubyClass).append(": ").append(exception.getByteList()).append('>')
-        );
+        String exceptionStartStr = new StringBuilder(2 + rubyClass.length() + 2).append("#<").append(rubyClass).append(": ").toString();
```
Comment (kares) [inline core/src/main/java/org/jruby/RubyException.java]: you should nuke the `StringBuilder` here - it shouldn't be needed

Comment (jlahtinen) [reply core/src/main/java/org/jruby/RubyException.java]: I think nuking is done now.

Comment (kares) [reply core/src/main/java/org/jruby/RubyException.java]: the last round-trip (Ruby -> Java -> Ruby string) is redundant, but we can fix that later ...

**Comment by enebo** [comment]
Super cool.  I may change the signature to be ruby,string,irubyobject,string,irubyobject,string but thanks for this work.  We really appreciate it!

**Pull request merged**